core/commands/generate_command.rb

```diff
@@ -322,12 +327,14 @@ def make_node_params(node, box_params)
     params[:ssh_pty] = box_params['ssh_pty'] == 'true' unless box_params['ssh_pty'].nil?
     params.merge!(if params[:provider] == 'aws'
```

Comment:
The code that follows the `if` statement should be moved out of this method. I.e. this method should provide generic data that should be updated for concrete cases.